src/StatsdClient/RandomGenerator.cs

```diff
@@ -4,15 +4,26 @@ namespace StatsdClient
 {
     public class RandomGenerator : IRandomGenerator
     {
-        readonly Random _random;
+        [ThreadStatic]
+        static Random _random;
+
+        private static Random Random
+        {
+            get
+            {
+                var random = _random;
+                if (random != null) return random;
+                return _random = new Random(Guid.NewGuid().GetHashCode());

```

Comment:
What about removing the `ThreadStatic` attribute (but keeping the private field static) and just letting all the threads share access to it?